crates/ruff_workspace/src/configuration.rs

```diff
@@ -59,64 +59,37 @@ pub struct RuleSelection {
 
 #[derive(Debug, Default)]
 pub struct Configuration {
-    pub rule_selections: Vec<RuleSelection>,
-    pub per_file_ignores: Option<Vec<PerFileIgnore>>,
-
-    pub allowed_confusables: Option<Vec<char>>,
-    pub builtins: Option<Vec<String>>,
+    // Global options
     pub cache_dir: Option<PathBuf>,
-    pub dummy_variable_rgx: Option<Regex>,
-    pub exclude: Option<Vec<FilePattern>>,
+    pub output_format: Option<SerializationFormat>,
+    pub fix: Option<bool>,
+    pub fix_only: Option<bool>,
+    pub show_fixes: Option<bool>,
+    pub show_source: Option<bool>,
+    pub required_version: Option<Version>,
+    pub preview: Option<PreviewMode>,
     pub extend: Option<PathBuf>,
+
+    // File resolver options
+    pub exclude: Option<Vec<FilePattern>>,
     pub extend_exclude: Vec<FilePattern>,
     pub extend_include: Vec<FilePattern>,
-    pub extend_per_file_ignores: Vec<PerFileIgnore>,
-    pub external: Option<Vec<String>>,
-    pub fix: Option<bool>,
-    pub fix_only: Option<bool>,
     pub force_exclude: Option<bool>,
-    pub output_format: Option<SerializationFormat>,
-    pub ignore_init_module_imports: Option<bool>,
     pub include: Option<Vec<FilePattern>>,
-    pub line_length: Option<LineLength>,
-    pub logger_objects: Option<Vec<String>>,
-    pub namespace_packages: Option<Vec<PathBuf>>,
-    pub preview: Option<PreviewMode>,
-    pub required_version: Option<Version>,
     pub respect_gitignore: Option<bool>,
-    pub show_fixes: Option<bool>,
-    pub show_source: Option<bool>,
-    pub src: Option<Vec<PathBuf>>,
-    pub tab_size: Option<TabSize>,
+
+    // Generic python options settings
+    pub builtins: Option<Vec<String>>,
+    pub namespace_packages: Option<Vec<PathBuf>>,
     pub target_version: Option<PythonVersion>,
-    pub task_tags: Option<Vec<String>>,
+    pub src: Option<Vec<PathBuf>>,
     pub typing_modules: Option<Vec<String>>,
```

Comment:
Hmm. I think `typing_modules` and `logger_objects` should arguably be treated the same. They're both used to mark modules or objects for multi-file analysis. E.g., if a module is in `typing_modules`, we assume it's exporting stuff from the standard library `typing` module. If an object is in `logger_objects`, we assume it's a `logging.Logger`.

I think these can probably both be part of the `lint` section.